tests/test_builder_local_image.py

```diff
@@ -0,0 +1,66 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import os
+import pytest
+import subprocess
+
+from galaxy_importer.ansible_test import builders
+
+
+@pytest.fixture
+def build():
+    return builders.local_image_build.Build('/file/path/to/archive.tar.gz', 'name')
+
+
+@pytest.fixture
+def tmp_file():
+    try:
+        dir = os.path.dirname(os.path.dirname(__file__))
+        tmp_file = os.path.join(dir, 'galaxy_importer', 'namespace-name-0.0.1.tar.gz')
+        yield tmp_file
+    finally:
+        os.remove(tmp_file)
+
+
+def test_build_image(mocker, tmp_file):
+    with open(tmp_file, 'w') as f:
+        f.write('file contents go here')
+        f.flush()
+        build = builders.local_image_build.Build(
+            filepath=tmp_file,
+            collection_name='namespace-name-version'
+        )
+        mocker.patch.object(build, '_build_dockerfile')
+        mocker.patch.object(build, '_copy_collection_file')
+        mocker.patch.object(build, '_build_image_with_artifact')
```

Comment:
These end up not mocking the methods called, see [gist](https://gist.github.com/awcrosby/f9144e83b6f266f3c271c88a970da468
)
`_build_dockerfile()` should probably have it's own test method.